### PR TITLE
feat(logging): Implement centralized logging with loguru

### PR DESCRIPTION
**Pull request by briansumma**
## Centralized Logging Implementation

This PR introduces a centralized logging system for the Pythonik SDK using the loguru library.

### Changes

1. **New Module: `_logger.py`**
   - Creates a single source of truth for logging configuration
   - Configures loguru with a stderr handler by default
   - Respects `LOGURU_LEVEL` environment variable with "INFO" as default

2. **Updates to Base Classes**
   - Replaces print statements with structured logging in `Spec.parse_response` and `Spec.send_request`
   - Provides formatted output for better debugging

3. **Import Standardization**
   - Updates imports in affected modules to use the centralized logger
   - Ensures consistent logging approach across the codebase

4. **Comprehensive Test Suite**
   - Tests for core logger configuration and environment variable handling
   - Tests for verification of logging in the base Spec class
   - Tests for import patterns across modules
   - Integration tests for end-to-end validation

### Benefits

- **Improved Debugging**: Structured logs with configurable verbosity
- **Environment Control**: Adjust log levels via environment variables
- **Consistent Interface**: Standardized logging approach throughout the codebase
- **Future Extensibility**: Foundation for more advanced logging features

### Testing

All tests pass as shown in the attached pytest logs. The implementation has been thoroughly tested for both functionality and edge cases.

### Usage Examples

Configure log level via environment variable:
```bash
LOGURU_LEVEL=DEBUG python your_script.py
```

Default level is INFO, which keeps output clean while still showing important information.

### Notes for Reviewers

- The changes are minimally invasive and maintain backward compatibility
- No changes to API interfaces or behavior
- Only debug-level logging has been implemented to avoid excessive output
- Tests have been designed to verify behavior rather than implementation details

### CODE REVIEW

File: pythonik/specs/base.py [hunk 1]
```diff
@@ -48,7 +49,7 @@ def parse_response(response: Response, model: Optional[Type[BaseModel]] = None)
         """
         # try to populate the model
         if response.ok:
-            print(response.text)
+            logger.debug(response.text)
```
Comment (nsa-brant) [inline pythonik/specs/base.py:52]: thanks for doing this, we need to get rid of the noise in pythonik...

File: pythonik/_logger.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import sys
```
Comment (nsa-brant) [inline pythonik/_logger.py:1]: Thanks for taking a swing at this, id prefer though if we followed loguru's suggested way of doing things here:

https://github.com/Delgan/loguru?tab=readme-ov-file#suitable-for-scripts-and-libraries

If i understand that doc, we'd disable our library logs by default and allow the user to opt in to enabling it

**Comment by briansumma** [comment]
The logger from pythonik._logger should match the same behavior as the logger from loguru but instead of the default level being DEBUG, it is INFO. You can still change the log level in the [environment](https://loguru.readthedocs.io/en/stable/api/logger.html) like you do the default logger, e.g., `export LOGURU_LEVEL=DEBUG`, `export LOGURU_FORMAT="{time} - {message}"`, so I don't think it negates that document link "Suitable for scripts and libraries" you sent. 

I think the only change would be to import the logger from pythonik/_logger.py instead of directly from loguru, i.e.,

```python
from pythonik._logger import logger

config = {
    "handlers": [
        {"sink": sys.stdout, "format": "{time} - {message}"},
        {"sink": "file.log", "serialize": True},
    ],
    "extra": {"user": "someone"}
}
logger.configure(**config)

...
```

**Comment by nsa-brant** [comment]
Closing as part of the pythonik 2.0 reset — the package is being replaced by a fully generated SDK (see #102). The 1.x line lives on the `maintenance/1.x` branch; if this change is still needed there, please reopen against that branch.
